cogs/displayinfo.py

```diff
@@ -33,85 +33,113 @@ class DisplayInfo(commands.Cog):
     async def on_member_update(self, mem_before:discord.Member, mem_after:discord.Member):
+        self.current_guild_name = mem_before.guild.name
+        
         if mem_before.activity and mem_before.activity == mem_after.activity:
             return
-    
-        if self.current_guild_name != mem_before.guild.name:
-            if mem_before.guild.name not in db.list_collection_names():
-                newCollection = db[mem_before.guild.name]  # If the collection doesn't exist, create a new one for the guild/server
-                self.current_guild_name = mem_before.guild.name
-                self.current_collection = newCollection               
-                newCollection.insert_one({}) # Create a new blank document in the new Collection
-                self.timers = self.current_collection.find_one() # Load blank document from the MongoDB collection
-            else:
-                self.current_collection = db[mem_before.guild.name] # If the collection exists, get the existing collection for the guild
-                self.current_guild_name = mem_before.guild.name
-                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+
+        if mem_before.status is discord.Status.offline and mem_after.status is discord.Status.online:
+            self.members_online.add(mem_after)
+            if self.timed_task.done():
+                self.timed_task = asyncio.create_task(self.timed_update())
+            print("added:", mem_after.name)
+        
+        elif mem_before.status is discord.Status.online and mem_after.status is discord.Status.offline:
+            self.members_online.remove(mem_before)
+            print("removed:", mem_before.name)
+            if len(self.members_online) <= 0:
+                self.timed_task.cancel()
+                try:
+                    await self.timed_task
+                except asyncio.CancelledError:
+                    print("task is now cancelled")
         
         # Track activity updates for members
         if not mem_before.activity and mem_after.activity: # activity starts/ongoing
-            if mem_after.name not in self.timers:
-                self.timers[mem_after.name] = {} # If the member is not already tracked, add them
-            if self.current_date not in self.timers[mem_after.name]:
-                self.timers[mem_after.name][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
-            if mem_after.activity.name not in self.timers[mem_after.name][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
-                self.timers[mem_after.name][self.current_date][mem_after.activity.name] = {
+            after_string_id = str(mem_after.id)
+            if after_string_id not in self.timers:
+                self.timers[after_string_id] = {} # If the member is not already tracked, add them
```

Comment:
still recommend making a ticket for updating these to defaultdicts. maybe something good for anthony so that we don't have so many of these checks in our code with `if <something> not in ...`